kart/sqlalchemy/create_engine.py

```diff
@@ -18,6 +18,21 @@
 L = logging.getLogger("kart.sqlalchemy.create_engine")
 
 
+def sqlite_engine(path):
+    """
+    An engine for non-spatial, non-GPKG sqlite databases.
+    """
+
+    def _on_connect(pysqlite_conn, connection_record):
+        pysqlite_conn.isolation_level = None
+        dbcur = pysqlite_conn.cursor()
+        dbcur.execute("PRAGMA foreign_keys = ON;")
+
+    engine = sqlalchemy.create_engine(f"sqlite:///{path}", module=sqlite)
```

Comment:
This should handle running in a read-only repository where we can't create or write to the DB.